routers/api/v1/api.go

```diff
@@ -1401,6 +1408,9 @@ func Routes() *web.Router {
 			}, repoAssignment(), checkTokenPublicOnly())
 		}, tokenRequiresScopes(auth_model.AccessTokenScopeCategoryRepository))
 
+		// Artifacts direct download endpoint authenticates via signed url
+		m.Get("/repos/{username}/{reponame}/actions/artifacts/{artifact_id}/zip/raw", repoAssignment(), repo.DownloadArtifactRaw)
```

Comment:
Before, it doesn't use `repoAssignment`.

* If there is no strong reason to skip `repoAssignment`, I think it is needed.
* If there is a strong reason to skip `repoAssignment`, it needs comments about why it must be skipped.